### PR TITLE
Templating: Makes so $__searchFilter can be used as part of regular expression

### LINKED ISSUE #20006: Template variable query $__searchFilter tweaks
There are some issues with using this with Graphite tag values queries:

tag_values(host, namespace=os, name=cpu.percent.idle.g, host=~^(.*?)$__searchFilter)

Due to the fact that in this context __searchFilter is in a regex context not a normal Graphite glob/wildcard context

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Next iteration of $__searchFilter adds the possibility to use it in regular expressions like the `tag_values` query in Graphite.

**Which issue(s) this PR fixes**:
Fixes #20006

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/templating/specs/variable.test.ts [hunk 1]
```diff
@@ -92,84 +93,164 @@ describe('containsSearchFilter', () => {
...
-        const quoteLiteral = false;
-
-        const result = interpolateSearchFilter({
-          query,
-          options,
-          wildcardChar,
-          quoteLiteral,
-        });
+  describe(`when called with a query with \$\{${SEARCH_FILTER_VARIABLE}:regex\}`, () => {
+    it('then it should return false', () => {
```
Comment (torkelo) [inline public/app/features/templating/specs/variable.test.ts]: should return true?

File: public/app/features/templating/specs/variable.test.ts [hunk 2]
```diff
@@ -92,84 +93,164 @@ describe('containsSearchFilter', () => {
...
-      const quoteLiteral = false;
-
-      const result = interpolateSearchFilter({
-        query,
-        options,
-        wildcardChar,
-        quoteLiteral,
-      });
+  describe(`when called with a query with [[${SEARCH_FILTER_VARIABLE}]]`, () => {
+    it('then it should return false', () => {
```
Comment (torkelo) [inline public/app/features/templating/specs/variable.test.ts]: ture?

**Review by torkelo — APPROVED** [review]
Tested this with Graphite with normal and tag_values queries, works great. Tested with $__searchFilter and ${__searchFilter} syntax and with ${__searchFilter:regex}

File: public/app/features/templating/specs/variable.test.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/templating/specs/variable.test.ts]: Good catch! Will look at them all

**Pull request merged**